distributed-tests/rackspace-server-manager.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python
+'''
+A script to create and delete the number of servers on Rackspace
+'''
+
+import pyrax
+import time
+import re
+import os
+import argparse
+import uuid
+
+
+def create_node(nova, counts):
+    flavor = nova.flavors.find(name='2 GB General Purpose v1')
+    image = nova.images.find(name='centos7-test')
+    pubkey = open('key.pub', 'r').read()
+    nova.keypairs.create('distkey', pubkey)
+    for count in range(int(counts)):
+        name = 'distributed-testing.'+str(uuid.uuid4())
+        node = nova.servers.create(name=name, flavor=flavor.id,
+                                   image=image.id, key_name='distkey')
+
+        while node.status == 'BUILD':
```

Comment:
Add a timeout. That is you'll only wait until a certain amount of time and after that you'll give up and exit with status code 1.

Reply:
We can add timeout in the jenkins job same as we do in other jobs.